gradle/buildScanInit.gradle

```diff
@@ -6,11 +10,19 @@ initscript {
         classpath "com.gradle:build-scan-plugin:1.12.1"
     }
 }
+
 if (!gradle.startParameter.systemPropertiesArgs.containsKey("disableScanPlugin")) {
     rootProject {
-        apply plugin: com.gradle.scan.plugin.BuildScanPlugin
-        buildScan {
-            server = "https://e.grdev.net"
+        pluginManager.withPlugin("com.gradle.build-scan") {
+            if (gradle.startParameter.taskNames in [["verifyTestFilesCleanup"], ["killExistingProcessesStartedByGradle"]]) {
```

Comment:
Why are we doing this check now in an init-script? Publish always should be activated for local builds which are not using the init script, too, shouldn't it?

Reply:
Because we have to deal with a `--scan` and non `--scan` scenario as explained below. This is not a check. Here it is about configuration order. In the previous version of the init script we have applied the plugin in the init script. Now we apply the plugin in the root build script. But we need to inject this behavior at a time when the plugin has not been aplied yet.

Reply:
The current behavior models the following behavior. 
1. If you run the build via `--scan` it automatically sets publishAlways. 
1. If you run the build without `--scan` and without the init script no build scan is produced. 
1. If you run the build with the init script, a build scan is always produced except when running certain tasks, when only one is produced in the case of a failure.

It is important for privacy reasons that we don't collect automatically a build scan for someone outside of our organization building Gradle. They can create one by using --scan or use their own init script (which should go to scans.gradle.com).

Does this makes sense?

Reply:
Makes sense.